crates/sui-core/src/consensus_handler.rs

```diff
@@ -646,238 +740,1639 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
+                    // (Note: we no lnoger need to worry about the previously deferred condition, since we are only
+                    // processing newly-received transactions at this time).
                     if parsed.transaction.kind.is_user_transaction() {
-                        self.epoch_store
-                            .set_consensus_tx_status(position, ConsensusTxStatus::Finalized);
-                        num_finalized_user_transactions[author] += 1;
+                        debug!(
+                            "Ignoring consensus transaction {:?} because of end of epoch",
+                            parsed.transaction.key()
+                        );
+                        continue;
                     }
-                    let kind = classify(&parsed.transaction);
-                    self.metrics
-                        .consensus_handler_processed
-                        .with_label_values(&[kind])
-                        .inc();
-                    self.metrics
-                        .consensus_handler_transaction_sizes
-                        .with_label_values(&[kind])
-                        .observe(parsed.serialized_len as f64);
-                    // UserTransaction exists only when mysticeti_fastpath is enabled in protocol config.
-                    if matches!(
-                        &parsed.transaction.kind,
-                        ConsensusTransactionKind::CertifiedTransaction(_)
-                            | ConsensusTransactionKind::UserTransaction(_)
-                    ) {
-                        self.last_consensus_stats
-                            .stats
-                            .inc_num_user_transactions(author);
+
+                    match &parsed.transaction.kind {
+                        ConsensusTransactionKind::UserTransaction(_)
+                        | ConsensusTransactionKind::CertifiedTransaction(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore capability notifications if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::CapabilityNotificationV2(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore end of publish messages if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::EndOfPublish(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore execution time observations if !should_accept_consensus_certs()
+                        // Note: we no longer have to check protocol_config.ignore_execution_time_observations_after_certs_closed()
+                        | ConsensusTransactionKind::ExecutionTimeObservation(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore jwk votes if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::NewJWKFetched(_, _, _) => continue,
```

Comment:
Maybe introducing a function to `ConsensusTransactionKind` called `is_randomness_system_tx`?

Reply:
i'm not sure i understand - those are not the only messages that are not ignored here. I changed this to an exhaustive match for better clarity and also to force a decision whenever we add new message types